### PR TITLE
Allow to provide array of commits to backport

### LINKED ISSUE #95: Allow backport of specific commit
That way we can use the compile bot to amend but still backport the commit changes only :)

`/backport 589fh4cds to stable16`

### PR DESCRIPTION
**Pull request by skjnldsv**
Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

```
/backport to stable18
/backport 1564a2c to stable18
/backport 1564a2c,52feca3 to stable18
```

fix #95

cc @christophwurst @tobiasKaminsky @juliushaertl

### CODE REVIEW

**Comment by skjnldsv** [comment]
Let me try this on a local repo

File: index.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ module.exports = app => {
       return
     }
 
-    //TODO filter same backport requests
+    // TODO: filter same backport requests
```
Comment (rullzer) [inline index.js:63]: crucial edit

Comment (skjnldsv) [reply index.js:63]: yes, it fixes my OCD :D

**Comment by skjnldsv** [comment]
Hum, doesn't work
https://github.com/skjnldsv/backport-test/pull/1

Seems that there is some issue
```
08:27:40.262Z DEBUG event: {"message":"Validation Failed","errors":[{"resource":"PullRequest","code":"custom","message":"No commits between stable1 and backport/1/stable1"}],"documentation_url":"https://developer.github.com/v3/pulls/#create-a-pull-request"} (id=c6fe2b00-7972-11ea-88c8-806fa02fc303)
  HttpError: {"message":"Validation Failed","errors":[{"resource":"PullRequest","code":"custom","message":"No commits between stable1 and backport/1/stable1"}],"documentation_url":"https://developer.github.com/v3/pulls/#create-a-pull-request"}
      at response.text.then.message (/rbd/pnpm-volume/703c4f72-6c78-41ba-94d5-a5ef2ed1cadd/node_modules/.registry.npmjs.org/@octokit/rest/15.18.3/node_modules/@octokit/rest/lib/request/request.js:72:19)
      at process._tickCallback (internal/process/next_tick.js:68:7)
08:27:40.262Z  WARN event: Backport to [object Object] failed (id=c6fe2b00-7972-11ea-88c8-806fa02fc303)

```

**Comment by skjnldsv** [comment]
Fixes incoming

**Comment by skjnldsv** [comment]
Wouhou :tada:

https://github.com/skjnldsv/backport-test/pull/3

**Comment by skjnldsv** [comment]
Fixed!
So of course if we amend, we cannot cherry-pick the amended commits: https://github.com/skjnldsv/backport-test/pull/5

Anyone knows a trick? Fresh clone don't have untracked commits in their history.
The only think I could think of is downloading from github & patching? :confounded:

**Pull request merged**